### DISCUSSION TITLE
Run Code After All Fixtures but Before Test

### DISCUSSION THREAD

**Discussion by cancan101** [pytest-dev/pytest#11092] [Q&A]
Is there some way to run certain code after all the fixtures have been created but before the test code itself is run?

For example I have a number of fixtures that create DB objects and would like to call `commit()` after they are created but before the test itself is run. I could just call commit at the end of each fixture but that slows down the test set up.

**Comment by The-Compiler** [comment]
One thing you could do is have a fixture what depends on the other ones:

```python
import pytest


@pytest.fixture
def create_a():
    print("create a")


@pytest.fixture
def create_b():
    print("create b")


@pytest.fixture
def db_objects(create_a, create_b):
    print("committing")


def test_db(db_objects):
    pass
```

If you wanted the db to be prepared like that for all tests, you could also make `db_objects` autouse.

**Reply by cancan101** [reply]
I would like to avoid having to list all of the dependent fixtures for a couple reasons:
- Different tests will have different requirements
- There are many, many fixtures and keeping the list up to date would be a challenge

`autouse` sounds interesting, but when would it run? ie would it run or after other fixtures? Or unspecified?

**Comment by RonnyPfannschmidt** [comment]
I believe that a runtest_setup trylast hook with some terrifying magic is necessary to sort this stuff out
